### PR TITLE
CVSL-465 Adding new job to remove AP conditions in PSS period

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/JobController.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  @Operation(
+    summary = "Job to remove AP conditions.",
+    description = "Triggers a job that removes AP conditions for all licences that are in PSS period and status equal to 'VARIATION_IN_PROGRESS' or 'VARIATION_SUBMITTED' or 'VARIATION_REJECTED' or 'VARIATION_APPROVED'. Requires ROLE_CVL_ADMIN.",
+    security = [SecurityRequirement(name = "ROLE_CVL_ADMIN")],
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Activation job executed.",
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/resource/JobController.kt:35]: needs to be updated to this job name

File: src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt [hunk 2]
```diff
@@ -289,25 +289,39 @@ class LicenceConditionService(
...
+    standardConditionIds: List<Long>,
+  ) {
     val username = SecurityContextHolder.getContext().authentication.name
     val currentUser = communityOffenderManagerRepository.findByUsernameIgnoreCase(username)
       ?: throw RuntimeException("Staff with username $username not found")
 
     // return all conditions except condition with submitted conditionIds
-    val revisedConditions = licenceEntity.additionalConditions.filter { conditionIds.indexOf(it.id) == -1 }
+    val revisedAdditionalConditions =
+      licenceEntity.additionalConditions.filter { additionalConditionIds.indexOf(it.id) == -1 }
```
Comment (andrewrlee) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt:303]: mmh.. would have thought you could have done `contains` here?

Comment (nishanthmahasamudram) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/LicenceConditionService.kt]: replaced

**Comment by nishanthmahasamudram** [comment]
already moved this files in another PR. So closing this.

**Pull request closed without merging**